anchore_engine/services/catalog/api/controllers/objects.py

```diff
@@ -94,3 +94,61 @@ def delete_object(bucket, archiveid):
         )
 
     return return_object, httpcode
+
+
+@flask_metrics.do_not_track()
+@authorizer.requires_account(with_types=INTERNAL_SERVICE_ALLOWED)
+def create_raw_object(bucket, archiveid, bodycontent):
+    httpcode = 500
+    try:
+        account_name = ApiRequestContextProxy.namespace()
+        obj_mgr = anchore_engine.subsys.object_store.manager.get_manager()
+
+        obj_mgr.put(account_name, bucket, archiveid, bodycontent)
+
+        my_svc = ApiRequestContextProxy.get_service()
+        if my_svc is not None:
+            resource_url = (
+                my_svc.service_record["base_url"]
+                + "/"
+                + my_svc.service_record["version"]
```

Comment:
will this _always_ work or is there any change this could cause a `KeyError` ?

Reply:
there is a case for keyerror. refactoring the try/except here so you'll see an explicit catch soon